frontend/src/components/course/components/edit-category-modal/edit-category-modal.tsx

```diff
@@ -0,0 +1,72 @@
+import {
+  CategoryGetAllItemResponseDto,
+  CourseUpdateCategoryRequestDto,
+  FC,
+  SelectorOptions,
+} from 'common/types/types';
+import { Button, Modal, Select } from 'components/common/common';
+import { getNameOf } from 'helpers/helpers';
+import { useAppDispatch, useAppForm, useMemo } from 'hooks/hooks';
+import { courseActions } from 'store/actions';
+import { courseUpdateCategory as courseUpdateCategoryValidationSchema } from 'validation-schemas/validation-schemas';
+
+import { getDefaultUpdateCourseCategoryPayload } from './common';
+import { getCategoriesOptions } from './helpers/helpers';
+
+type Props = {
+  courseId: number;
+  defaultCategoryId: number;
+  isModalOpen: boolean;
+  categories: CategoryGetAllItemResponseDto[];
+  onModalToggle: () => void;
+};
+
+const EditCategoryModal: FC<Props> = ({
+  courseId,
+  defaultCategoryId,
+  isModalOpen,
+  categories,
+  onModalToggle,
+}) => {
+  const dispatch = useAppDispatch();
+
+  const { control, errors, handleSubmit } =
+    useAppForm<CourseUpdateCategoryRequestDto>({
+      defaultValues: getDefaultUpdateCourseCategoryPayload(defaultCategoryId),
+      validationSchema: courseUpdateCategoryValidationSchema,
+    });
+
+  const categoriesOptions = useMemo<SelectorOptions<string>[]>(() => {
+    return getCategoriesOptions(categories);
+  }, [categories]);
+
+  const onSubmit = (payload: CourseUpdateCategoryRequestDto): void => {
```

Comment:
```suggestion
  const handleSubmit = (payload: CourseUpdateCategoryRequestDto): void => {
```

Reply:
There wil be name error, so I renamed as handleModalSubmit 👌